File: apps/executive-dashboard/src/types/country.ts
/**
 * Country Types — exact mirror of schemas/country.py + schemas/gdp.py
 *
 * Source: services/intelligence-api/app/schemas/country.py
 */

import type { LanguageVariant, GCCCountryCode, GDPComponentCode } from "./scenario";

// ─── GDP Impact (from gdp.py) ───────────────────────────────────

export interface GDPComponentImpact {
  component: GDPComponentCode;
  impact_score: number; // -1 to 1
  direction: "positive" | "negative" | "neutral";
  confidence: number; // 0-1
  drivers: string[];
  explanation: LanguageVariant;
}

export interface GDPImpactResult {
  scenario_id: string;
  country_code: GCCCountryCode;
  components: GDPComponentImpact[];
  aggregate_impact: number; // -1 to 1
  explanation: LanguageVariant;
}

// ─── Country Impact ─────────────────────────────────────────────

export interface PublicSectorImpact {
  spending_pressure: number; // 0-100
  policy_sensitivity: number; // 0-100
  infrastructure_continuity: number; // 0-100
  regulatory_sensitivity: number; // 0-100
  strategic_priorities: string[];
  explanation: LanguageVariant;
}

export interface PrivateSectorImpact {
  operating_cost_pressure: number; // 0-100
  financing_pressure: number; // 0-100
  demand_pressure: number; // 0-100
  investment_sentiment: number; // 0-100
  startup_sensitivity: number; // 0-100
  explanation: LanguageVariant;
}

export interface CountryImpact {
  country_code: GCCCountryCode;
  macro_sensitivity: number; // 0-100
  gdp_impact: GDPImpactResult;
  public_sector: PublicSectorImpact;
  private_sector: PrivateSectorImpact;
  risk_level: "critical" | "high" | "elevated" | "moderate" | "stable";
  narrative: LanguageVariant;
  confidence: number; // 0-1
}

File: services/intelligence-api/app/schemas/country.py
"""Country impact schemas — GCC country-level intelligence with public/private split."""
from pydantic import BaseModel, Field
from .scenario import GCCCountryCode, LanguageVariant
from .gdp import GDPImpactResult


class PublicSectorImpact(BaseModel):
    """Government and sovereign sector impact assessment."""
    spending_pressure: float = Field(ge=0.0, le=100.0)
    policy_sensitivity: float = Field(ge=0.0, le=100.0)
    infrastructure_continuity: float = Field(ge=0.0, le=100.0)
    regulatory_sensitivity: float = Field(ge=0.0, le=100.0)
    strategic_priorities: list[str]
    explanation: LanguageVariant


class PrivateSectorImpact(BaseModel):
    """Private sector and market-facing impact assessment."""
    operating_cost_pressure: float = Field(ge=0.0, le=100.0)
    financing_pressure: float = Field(ge=0.0, le=100.0)
    demand_pressure: float = Field(ge=0.0, le=100.0)
    investment_sentiment: float = Field(ge=0.0, le=100.0)
    startup_sensitivity: float = Field(ge=0.0, le=100.0)
    explanation: LanguageVariant


class CountryImpact(BaseModel):
    """Full country-level intelligence output."""
    country_code: GCCCountryCode
    macro_sensitivity: float = Field(ge=0.0, le=100.0)
    gdp_impact: GDPImpactResult
    public_sector: PublicSectorImpact
    private_sector: PrivateSectorImpact
    risk_level: str = Field(pattern="^(critical|high|elevated|moderate|stable)$")
    narrative: LanguageVariant
    confidence: float = Field(ge=0.0, le=1.0)
